backend/infrahub/core/diff/conflicts_identifier.py

```diff
@@ -0,0 +1,266 @@
+from infrahub.core.constants.database import DatabaseEdgeType
+from infrahub.core.schema_manager import SchemaManager
+from infrahub.core.timestamp import Timestamp
+
+from .coordinator import DiffCoordinator
+from .model.diff import BranchChanges, DataConflict, ModifiedPathType
+from .model.path import (
+    EnrichedDiffAttribute,
+    EnrichedDiffNode,
+    EnrichedDiffProperty,
+    EnrichedDiffRelationship,
+    EnrichedDiffSingleRelationship,
+)
+
+
+class ConflictsIdentifier:
+    def __init__(
+        self, diff_coordinator: DiffCoordinator, base_branch: Branch, diff_branch: Branch, schema_manager: SchemaManager
+    ) -> None:
+        self.base_branch = base_branch
+        self.diff_branch = diff_branch
+        self.diff_coordinator = diff_coordinator
+        self.schema_manager = schema_manager
+
+    async def get_conflicts(
+        self,
+        from_time: Timestamp | None = None,
+        to_time: Timestamp | None = None,
+    ) -> list[DataConflict]:
+        from_time = from_time or Timestamp(self.diff_branch.created_at)
+        to_time = to_time or Timestamp()
+        base_diff_root = await self.diff_coordinator.get_diff(
+            base_branch=self.base_branch, diff_branch=self.base_branch, from_time=from_time, to_time=to_time
+        )
+        branch_diff_root = await self.diff_coordinator.get_diff(
+            base_branch=self.base_branch, diff_branch=self.diff_branch, from_time=from_time, to_time=to_time
+        )
+
+        conflicts = []
+        base_node_map = {n.uuid: n for n in base_diff_root.nodes}
+        branch_node_map = {n.uuid: n for n in branch_diff_root.nodes}
+        common_node_uuids = set(base_node_map.keys()) & set(branch_node_map.keys())
+        for node_uuid in common_node_uuids:
+            base_node = base_node_map[node_uuid]
+            branch_node = branch_node_map[node_uuid]
+            conflicts.extend(self._build_node_conflicts(base_node=base_node, branch_node=branch_node))
+
+        return conflicts
+
+    def _build_node_conflicts(self, base_node: EnrichedDiffNode, branch_node: EnrichedDiffNode) -> list[DataConflict]:
+        conflicts: list[DataConflict] = []
+        if base_node.action != branch_node.action:
+            conflicts.append(self._add_node_conflict(base_node=base_node, branch_node=branch_node))
+        base_attribute_map = {a.name: a for a in base_node.attributes}
+        branch_attribute_map = {a.name: a for a in branch_node.attributes}
+        common_attribute_names = set(base_attribute_map.keys()) & set(branch_attribute_map.keys())
+        for attr_name in common_attribute_names:
+            base_attribute = base_attribute_map[attr_name]
+            branch_attribute = branch_attribute_map[attr_name]
+            conflicts.extend(
+                self._build_attribute_conflicts(
+                    branch_node=branch_node, base_attribute=base_attribute, branch_attribute=branch_attribute
+                )
```

Comment:
When changing a value of an attribute to the same in both branches this would be handled by the HAS_VALUE property right? I.e. it would not be marked as a conflict.

Reply:
that is correct